lib/src/component_declaration/flux_component.dart

```diff
@@ -76,46 +101,101 @@ abstract class FluxUiComponent<TProps extends FluxUiProps> extends UiComponent<T
 /// Use with the over_react transformer via the `@Component()` ([annotations.Component]) annotation.
 abstract class FluxUiStatefulComponent<TProps extends FluxUiProps, TState extends UiState>
     extends UiStatefulComponent<TProps, TState>
-    with _FluxComponentMixin<TProps>, BatchedRedraws {}
+    with _FluxComponentMixin<TProps>, BatchedRedraws {
+  // Redeclare these lifecycle methods with `mustCallSuper`, since `mustCallSuper` added to methods within
+  // mixins doesn't work. See https://github.com/dart-lang/sdk/issues/29861
+
+  @mustCallSuper
+  @override
+  // Ignore this warning to work around https://github.com/dart-lang/sdk/issues/29860
+  // ignore: must_call_super
+  void componentWillMount();
+
+  @mustCallSuper
+  @override
+  // Ignore this warning to work around https://github.com/dart-lang/sdk/issues/29860
+  // ignore: must_call_super
+  void componentWillReceiveProps(Map prevProps);
+
+  @mustCallSuper
+  @override
+  void componentDidUpdate(Map prevProps, Map prevState);
+
+  @mustCallSuper
+  @override
+  void componentWillUnmount();
+}
 
 /// Helper mixin to keep [FluxUiComponent] and [FluxUiStatefulComponent] clean/DRY.
 ///
 /// Private so it will only get used in this file, since having lifecycle methods in a mixin is risky.
-abstract class _FluxComponentMixin<TProps extends FluxUiProps> implements BatchedRedraws {
-  TProps get props;
-
+abstract class _FluxComponentMixin<TProps extends FluxUiProps> implements BatchedRedraws, UiComponent<TProps> {
   /// List of store subscriptions created when the component mounts.
   ///
   /// These subscriptions are canceled when the component is unmounted.
-  List<StreamSubscription> _subscriptions = [];
+  List<StreamSubscription> _subscriptions;
+
+  bool get _areStoreHandlersBound => _subscriptions != null;
+
+  /// Subscribe to all applicable stores.
+  ///
+  /// [Store]s returned by [redrawOn] will have their triggers mapped directly to this components
+  /// redraw function.
+  ///
+  /// [Store]s included in the [getStoreHandlers] result will be listened to and wired up to their
+  /// respective handlers.
+  void _bindStoreHandlers() {
+    if (_areStoreHandlersBound) {
+      throw new StateError('Store handlers are already bound');
```

Comment:
The only case this would be hit is if someone doesn't add a super-call.

I wasn't sure if it was worth adding a test around, but I can add one if you'd like.